doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -1276,7 +1149,7 @@ in support libraries of plug-ins.
 Exposing constants to OCaml libraries
 ````````````````````````````````````````````````````````````````
 
-.. cmd:: Register @qualid__1 as @qualid__2
+.. cmd:: Register @qualid as @qualid
```

Comment:
Please undo this change. The grammar checker should be able to discard indices before verifying that a `cmd` corresponds to the actual grammar and should not change them if it does. This is really important because in case of repeated nonterminals, it is quite useful to be able to distinguish them in the description.

Reply:
Sure, but if the line doesn't match exactly after removing subscripts, there's a message.  The developer will have to do the update manually and perhaps adjust the subscripts.

Reply:
If the line doesn't match exactly after removing subscripts, the tool could update the line with the correct replacement, with no subscripts. The developer can then intervene manually to add subscripts if they are useful. They have to intervene anyway to fix the description of the command so that's not a big deal.

Reply:
It could.  We can decide when we're closer to adding the tool to the CI build.  There will definitely be messages the developer has to deal with manually.  For example, there will be a message like "nonterminal 'foo' not included in the rst files".  In most cases, only the developer knows where that nonterminal should appear.  And if it shouldn't appear, there should be an edit that removes it from `orderedGrammar`.  IMHO better to draw the developers attention to the details rather than rely on them reading the updates carefully (or for the review process to catch a glitch).

Reply:
This might not be trivial to figure out what is the exact thing to put in there if the tool doesn't update it automatically. But the tool could still print a warning when things have been changed in the doc, with the files and line numbers, telling the developer to review and adapt.

Reply:
This is how it works now when it doesn't match (note "as" -> "as2"):

```
Warning: doc/sphinx/proof-engine/vernacular-commands.rst line 1139: NO MATCH `Register @qualid__1 as @qualid__2`
Warning: doc/sphinx/proof-engine/vernacular-commands.rst line 1139: BEST `Register @qualid as2 @qualid`
```

Reply:
OK, seems good enough :+1:

Reply:
One further daydream: if messages are formatted correctly and the build is run from within the developer's IDE, many IDEs parse the location of each message and provide a command to jump to file/line # for the next message.  But different IDEs may expect different formatting.  At the moment, doc_grammar doesn't have an easy way to report the line number for `common.edit_mlg` messages.  And I don't think you can usefully associate a "nonterminal XXX doesn't appear the rsts" with a specific line in the input.